python/lsst/analysis/tools/analysisPlots/astrometricResidualsPlots.py

```diff
@@ -0,0 +1,151 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = (
+    "StellarAstrometricResidualsRAFocalPlanePlot",
+    "StellarAstrometricResidualsDecFocalPlanePlot",
+    "StellarAstrometricResidualsSkyPlot",
+)
+
+from ..actions.plot.focalPlanePlot import FocalPlanePlot
+from ..actions.plot.skyPlot import SkyPlot
+from ..actions.vector import (
+    BandSelector,
+    ConvertUnits,
+    DownselectVector,
+    LoadVector,
+    MagColumnNanoJansky,
+    ResidualWithPerGroupStatistic,
+    SnSelector,
+    ThresholdSelector,
+)
+from ..interfaces import AnalysisPlot
+
+
+class StellarAstrometricResidualsBase(AnalysisPlot):
+    """Plot mean astrometric residuals.
+
+    The individual source measurements are grouped together by object index
+    and the per-group magnitude is computed. The residuals between the
+    individual sources and these magnitudes are then used to construct a plot
+    showing the mean residual as a function of the focal-plane or sky position.
+    """
+
+    def setDefaults(self):
+        super().setDefaults()
+
+        # Apply per-source selection criteria
+        self.prep.selectors.bandSelector = BandSelector()
+
+        self.process.buildActions.mags = MagColumnNanoJansky(vectorKey="psfFlux")
+        self.process.buildActions.residual = ConvertUnits()
+        self.process.buildActions.residual.inUnit = "degree"
+        self.process.buildActions.residual.outUnit = "marcsec"
+
+        self.process.buildActions.residual.buildAction = ResidualWithPerGroupStatistic()
+
+        self.process.buildActions.x = LoadVector(vectorKey="x")
+        self.process.buildActions.y = LoadVector(vectorKey="y")
+
+        self.process.buildActions.detector = LoadVector(vectorKey="detector")
+
+        self.process.filterActions.x = DownselectVector(vectorKey="x")
+        self.process.filterActions.x.selector = ThresholdSelector(
+            vectorKey="mags",
+            op="le",
+            threshold=24,
+        )
+        self.process.filterActions.y = DownselectVector(
+            vectorKey="y", selector=self.process.filterActions.x.selector
+        )
+        self.process.filterActions.z = DownselectVector(
+            vectorKey="residual", selector=self.process.filterActions.x.selector
+        )
+        self.process.filterActions.detector = DownselectVector(
+            vectorKey="detector", selector=self.process.filterActions.x.selector
+        )
+
+        self.process.buildActions.statMask = SnSelector()
+        self.process.buildActions.statMask.threshold = 0
+        self.process.buildActions.statMask.fluxType = "psfFlux"
+
+
+class StellarAstrometricResidualsSkyPlot(StellarAstrometricResidualsBase):
```

Comment:
Yeah, I mostly just put this in as an example, and I didn't add it to the pipeline file, because I don't think these plots are very interesting. However, I'll add Dec in case it becomes useful down the line.